src/Backend/App.vala

```diff
@@ -159,16 +155,18 @@ public class Slingshot.Backend.App : Object {
                 count_visible = prop_value.get_boolean ();
             }
         }
-
-        unity_update_info ();
     }
 
     public void unity_reset () {
         unity_sender_name = null;
         count_visible = false;
         current_count = 0;
+    }
 
-        unity_update_info ();
+    public void remove_launcher_entry (string sender_name) {
```

Comment:
I don't really like the introduction of this method. I liked the previous approach of having the `remove_launcher_entry` in the `SlingshotView` and having it compare all the senders on the fly. This method can be only called with the right `sender_name` which could cause some confusion when dealing with it later. What I mean here, is putting the sender name as an argument that is some kind of "key" rather than something that would change the way a method would execute.